app/controllers/api/v1/accounts_controller.rb

```diff
@@ -47,6 +47,8 @@ def follow
     options = @account.locked? || current_user.account.silenced? ? {} : { following_map: { @account.id => { reblogs: follow.show_reblogs?, notify: follow.notify?, languages: follow.languages } }, requested_map: { @account.id => false } }
 
     render json: @account, serializer: REST::RelationshipSerializer, relationships: relationships(**options)
+  rescue FollowService::SelfFollowError
```

Comment:
Is this ever actually raised?

Reply:
Good catch. Let me go over it and make sure that it is raised 👍🏿

Reply:
One idea here -- could we re-use `Mastodon::ValidatorError` (with a custom string)?

I think this could keep the entire check at the controller level, would match existing style, and would let you remove all the new error classes, along with the changes to the service and service spec.

For an example of what I mean - look at `api/v1/filters#update` and how it raises that class (or `check_accounts_limit` in this controller, though it does not have a custom message).

I suspect you could keep your request spec changes here, but move the actual check to a `before_action` in the controller (since it knows what accounts are relevant before passing in).